loading.bs

```diff
@@ -0,0 +1,949 @@
+
+<h3 algorithm id="validating-signature">Validating a signature</h3>
+
+An [=exchange signature=] |signature| <dfn for="exchange signature" lt="is
+valid|is not valid">is valid</dfn> for a [=byte sequence=] |headerBytes| if the
+following steps return `valid`:
+
+1. Let |clockSkew| be the uncertainty in the UA's estimate of the current time
+    caused by clock skew on the client. The UA MAY set this to 0 or use a more
+    sophisticated estimate.
+1. If the UA's estimate of the current time is more than |clockSkew| before
+    |signature|'s [=exchange signature/date=], return "untrusted".
+
+    Note: We take estimated clock skew into account when checking the
+    signature's [=exchange signature/date=] because we want well-behaved servers
+    to use the time they created the signature, but if they immediately start
+    serving that signature, and skewed clients don't try to correct for their
+    skew, those clients will reject the signature.
+1. If the UA's estimate of the current time is after |signature|'s [=exchange
+    signature/expiration time=], return "untrusted".
+
+    Note: We use the client's best guess of the current time to check the
+    [=exchange signature/expiration time=] so that attackers trying to get an
+    exchange trusted for longer, are constrained to modify the client's clock
+    and can't also attack its estimate of its skew.
+1. Let |message| be the [=b1 signed message=] for |signature| and |headerBytes|.
+1. Let |publicKey| be the [=certificate/public key=] of |parsedSignature|'s
+    [=exchange signature/certificate chain=]'s [=certificate chain/leaf=]. If
+    the certificate can't be parsed enough to find this public key, return
+    `invalid`.
+1. If |publicKey|'s [=public key/algorithm=] is not [=id-ecPublicKey=] on the
+    [=secp256r1=] named curve, return `invalid`.
+1. If |parsedSignature|'s [=exchange signature/signature=] is not a valid
+    signature of |message| by |publicKey| using the [=ecdsa_secp256r1_sha256=]
+    algorithm, return `invalid`.
+1. Return `valid`.
+
+<h3 algorithm id="cross-origin-trust">Cross-origin trust</h3>
+
+A [=exchange signature/is valid|valid=] [=exchange signature=] |signature| <dfn
+for="exchange signature" lt="establishes cross-origin trust|does not establish
+cross-origin trust">establishes cross-origin trust</dfn> in an [=exchange=]
+|exchange| if the following steps return "trusted":
+
+1. Let |requestUrl| be |exchange|'s [=exchange/request=]'s [=request/url=].
+1. If |signature|'s [=exchange signature/validityUrl=]'s [=url/origin=] is not
+    [=same origin=] with |requestUrl|'s [=url/origin=], return "untrusted".
+1. If |exchange|'s [=exchange/request=]'s [=request/header list=] includes a
+    [=stateful request header=] or |exchange|'s [=exchange/response=]'s
+    [=response/header list=] includes a [=stateful response header=], return
+    "untrusted".
+1. If |signature|'s [=exchange signature/expiration time=] is more than 604800
```

Comment:
I check the length of the validity window in cross-origin trust rather than while parsing the signature so that other uses of signatures can define their own maximum validities. e.g. maybe the Play store wants to sign that it's audited an app and have that be valid for more than a week.